wkcuber/api/TiffData/TiffMag.py

```diff
@@ -146,30 +264,90 @@ def close(self):
         return
 
     def calculate_relevant_slices(self, offset, shape):
-        for z in range(offset[2] + 1, offset[2] + shape[2] + 1):
-            yield tuple(
-                (z, offset[0:2], shape[0:2])
-            )  # return tuple of important z layers an the x-y offset (without z offset) and the size (without z length)
+        tile_size = self.header.tile_size
+
+        max_indices = tuple(i1 + i2 for i1, i2 in zip(offset, shape))
+
+        x_first_index = (
+            offset[0] // tile_size[0] if tile_size else None
+        )  # floor division
+        x_last_index = (
+            -(-max_indices[0] // tile_size[0]) if tile_size else None
```

Comment:
I changed it as you said because it is more readable.

I found the old solution here https://stackoverflow.com/questions/14822184/is-there-a-ceiling-equivalent-of-operator-in-python. This was by far the most upvoted answer. The benefits are that the method works better with larger numbers, but I guess this is not needed in this case.